### PR TITLE
Add attribute for k8s pod annotations

### PR DESCRIPTION
**Pull request by ChrsMark**
## Changes

This adds support for pod's annotations to k8s.pod resource.
ref: https://github.com/open-telemetry/semantic-conventions/pull/494#discussion_r1400848539

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/attributes-registry/k8s.md
Comment (dashpole) [reply docs/attributes-registry/k8s.md:23]: It still is listed as Recommended on this line

Comment (ChrsMark) [reply docs/attributes-registry/k8s.md]: Yeap, the registry file was missing the `omit_requirement_level`. It should be ok now.

**Review by dashpole — APPROVED** [review]

**Review by jinja2 — APPROVED** [review]

**Comment by ChrsMark** [comment]
@open-telemetry/specs-semconv-approvers @open-telemetry/specs-semconv-maintainers could you have a look please?

File: docs/resource/k8s.md
Comment (pyohannes) [inline docs/resource/k8s.md:95]: Maybe a nit, but I wonder if `annotations` and `labels` should be pluralized or not.

There's a similar pattern used by stable HTTP conventions, there the singular is used ([`http.response.header.<key>`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#common-attributes)).

It would make sense to settle on the singular here too (`k8s.pod.annotation.<key>` and `k8s.pod.label.<key>`), for consistency, and also because each attribute only refers to a single label or annotation.

Comment (ChrsMark) [reply docs/resource/k8s.md:95]: I don't have a strong preference for that so yeap maybe it's a good idea to align `annotations` and `labels` with what we have in `http.response.header.*`.

What @open-telemetry/semconv-k8s-approvers think about this?

Comment (TylerHelmuth) [reply docs/resource/k8s.md:95]: I think being consistent is important, so sounds like we should have `label` and `annotation`.

**Comment by ChrsMark** [comment]
Based on the discussion at https://github.com/open-telemetry/semantic-conventions/pull/573#discussion_r1417158430 I have updated this PR to make the annotations attribute singular.
If we agree on this and the PR is approved I will create a separate issue to change `k8s.pod.labels.*` and `container.labels.*` as well.

**Review by pyohannes — APPROVED** [review]

File: docs/resource/k8s.md
Comment (jinja2) [reply docs/resource/k8s.md]: It makes sense to drop the pluralization given the existing guidance. Once this lands, we'll need to update the k8sattr processor which uses `labels` and `annotations` at present.

Comment (ChrsMark) [reply docs/resource/k8s.md]: Thank's folks. Would you mind giving another review on the updated PR?
Happy to help with the implementation's side changes as well :).

**Review by AlexanderWert — APPROVED** [review]

**Review by mlunadia — APPROVED** [review]
LGTM

**Review by TylerHelmuth — APPROVED** [review]

File: docs/resource/k8s.md
Comment (trask) [reply docs/resource/k8s.md]: good catch 👍

**Review by trask — APPROVED** [review]

**Pull request merged**
